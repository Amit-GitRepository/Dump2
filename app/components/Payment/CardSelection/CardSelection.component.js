/* CardSelection: shows list of cards and option to pay with new card*/

import jcb from '../../../assets/jcb.png';
import mastercard from '../../../assets/mastercard.png';
import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './CardSelection.style';
import visa from '../../../assets/visa.png';
import {Icon, ISText, RadioButton} from '../../Shared';
import {Image, View} from 'react-native';
/* eslint-disable camelcase */

const mapCardBrandImage = {
  'Visa': visa,
  'JCB': jcb,
  'Mastercard': mastercard,
  'MasterCard': mastercard
};

class CardSelection extends Component {

  deleteCard = (cardId) => () => {
    this.props.deleteCard(cardId);
  }

  render () {
    const {isSelected, cardDetails, onSelect, isLast, deleteCard} = this.props;
    const {brand, name, last_digits, id} = cardDetails;
    return (
      <View style={[styles.container, isLast && styles.hideBorder]}>
        <View style={styles.textContainer}>
          <RadioButton isSelected={isSelected} onRadioSelect={onSelect}/>
          {brand && <Image source={mapCardBrandImage[brand]} style={styles.cardType}/>}
          <ISText type='SEMIBOLD' style={styles.cardName} numberOfLines={1} ellipsizeMode='tail'>{name}</ISText>
          <ISText type='SEMIBOLD' style={styles.cardNumber}>{last_digits}</ISText>
        </View>

        {deleteCard && <Icon size={20} name='delete' style={styles.deleteButton} onPress={this.deleteCard(id)}/>}
      </View>
    );
  }
}
CardSelection.defaultProps = {
  isSelected: false,
  isLast: false,
  cardDetails: {},
  onSelect: noop
};
CardSelection.propTypes = {
  deleteCard: PropTypes.func,
  isLast: PropTypes.bool,
  onSelect: PropTypes.func,
  cardDetails: PropTypes.object,
  isSelected: PropTypes.bool
};
export default CardSelection;
