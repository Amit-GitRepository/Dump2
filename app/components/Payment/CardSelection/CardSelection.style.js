import {colors, fonts} from '../../../themes/constants.styles';

export default {
  container: {
    flexDirection: 'row',
    paddingVertical: 10,
    alignItems: 'center',
    borderBottomWidth: 1,
    justifyContent: 'space-between',
    borderColor: colors.PRIMARY_BG_SEPARATOR
  },
  cardType: {
    flex: 0,
    marginLeft: 10
  },
  textContainer: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  cardName: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    paddingHorizontal: 5,
    marginLeft: 10,
    maxWidth: 130
  },
  deleteButton: {
    alignSelf: 'center',
    color: colors.PRIMARY_ACTIONABLE,
    marginRight: 10
  },
  cardNumber: {
    fontSize: fonts.FONT_SIZE_NORMAL,
    color: colors.PRIMARY_ACTIONABLE
  },
  hideBorder: {
    borderBottomWidth: 0
  }
};
