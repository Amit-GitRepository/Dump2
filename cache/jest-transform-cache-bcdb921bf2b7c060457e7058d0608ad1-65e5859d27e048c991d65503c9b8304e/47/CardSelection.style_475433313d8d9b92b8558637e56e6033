Object.defineProperty(exports,"__esModule",{value:true});var cov_2dx5etunzs=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardSelection/CardSelection.style.js',hash='742407331eb152f7e36d6cd1607b88508a1116c4',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardSelection/CardSelection.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
flexDirection:'row',
paddingHorizontal:5,
paddingVertical:10,
alignItems:'center',
borderBottomWidth:1,
borderColor:_constants.colors.PRIMARY_BG_SEPARATOR},

cardType:{
flex:0,
marginLeft:10},

cardName:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM,
paddingHorizontal:5,
marginLeft:10},

cardNumber:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
color:_constants.colors.PRIMARY_ACTIONABLE},

hideBorder:{
borderBottomWidth:0},

deleteButton:{
marginLeft:15}};