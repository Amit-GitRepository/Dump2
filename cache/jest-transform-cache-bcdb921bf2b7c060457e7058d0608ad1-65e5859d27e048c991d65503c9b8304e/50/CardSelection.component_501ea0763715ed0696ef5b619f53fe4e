Object.defineProperty(exports,"__esModule",{value:true});var cov_2dhks0odzo=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardSelection/CardSelection.component.js',hash='7f7a107a7951d9eaa36f9f7cc6f5e8b107b41aa7',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardSelection/CardSelection.component.js',statementMap:{'0':{start:{line:14,column:26},end:{line:18,column:1}},'1':{start:{line:22,column:27},end:{line:24,column:3}},'2':{start:{line:23,column:4},end:{line:23,column:34}},'3':{start:{line:27,column:68},end:{line:27,column:78}},'4':{start:{line:28,column:43},end:{line:28,column:54}},'5':{start:{line:29,column:4},end:{line:37,column:6}},'6':{start:{line:40,column:0},end:{line:45,column:2}},'7':{start:{line:46,column:0},end:{line:52,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:22,column:15},end:{line:22,column:16}},loc:{start:{line:22,column:27},end:{line:24,column:3}},line:22},'1':{name:'(anonymous_1)',decl:{start:{line:22,column:27},end:{line:22,column:28}},loc:{start:{line:22,column:33},end:{line:24,column:3}},line:22},'2':{name:'(anonymous_2)',decl:{start:{line:26,column:2},end:{line:26,column:3}},loc:{start:{line:26,column:12},end:{line:38,column:3}},line:26}},branchMap:{'0':{loc:{start:{line:30,column:38},end:{line:30,column:65}},type:'binary-expr',locations:[{start:{line:30,column:38},end:{line:30,column:44}},{start:{line:30,column:48},end:{line:30,column:65}}],line:30},'1':{loc:{start:{line:32,column:9},end:{line:32,column:84}},type:'binary-expr',locations:[{start:{line:32,column:9},end:{line:32,column:14}},{start:{line:32,column:18},end:{line:32,column:84}}],line:32},'2':{loc:{start:{line:35,column:9},end:{line:35,column:118}},type:'binary-expr',locations:[{start:{line:35,column:9},end:{line:35,column:19}},{start:{line:35,column:23},end:{line:35,column:118}}],line:35}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0},f:{'0':0,'1':0,'2':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();

var _jcb=require('../../../assets/jcb.png');var _jcb2=_interopRequireDefault(_jcb);
var _mastercard=require('../../../assets/mastercard.png');var _mastercard2=_interopRequireDefault(_mastercard);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _CardSelection=require('./CardSelection.style');var _CardSelection2=_interopRequireDefault(_CardSelection);
var _visa=require('../../../assets/visa.png');var _visa2=_interopRequireDefault(_visa);
var _Shared=require('../../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}


var mapCardBrandImage=(cov_2dhks0odzo.s[0]++,{
'Visa':_visa2.default,
'JCB':_jcb2.default,
'Mastercard':_mastercard2.default});var


CardSelection=function(_Component){_inherits(CardSelection,_Component);function CardSelection(){var _ref;var _temp,_this,_ret;_classCallCheck(this,CardSelection);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=CardSelection.__proto__||Object.getPrototypeOf(CardSelection)).call.apply(_ref,[this].concat(args))),_this),_this.

deleteCard=function(cardId){cov_2dhks0odzo.f[0]++;cov_2dhks0odzo.s[1]++;return function(){cov_2dhks0odzo.f[1]++;cov_2dhks0odzo.s[2]++;
_this.props.deleteCard(cardId);
};},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(CardSelection,[{key:'render',value:function render()

{cov_2dhks0odzo.f[2]++;var _ref2=(cov_2dhks0odzo.s[3]++,
this.props),isSelected=_ref2.isSelected,cardDetails=_ref2.cardDetails,onSelect=_ref2.onSelect,isLast=_ref2.isLast,deleteCard=_ref2.deleteCard;var _ref3=(cov_2dhks0odzo.s[4]++,
cardDetails),brand=_ref3.brand,name=_ref3.name,last_digits=_ref3.last_digits,id=_ref3.id;cov_2dhks0odzo.s[5]++;
return(
_react2.default.createElement(_reactNative.View,{style:[_CardSelection2.default.container,(cov_2dhks0odzo.b[0][0]++,isLast)&&(cov_2dhks0odzo.b[0][1]++,_CardSelection2.default.hideBorder)]},
_react2.default.createElement(_Shared.RadioButton,{isSelected:isSelected,onRadioSelect:onSelect}),
(cov_2dhks0odzo.b[1][0]++,brand)&&(cov_2dhks0odzo.b[1][1]++,_react2.default.createElement(_reactNative.Image,{source:mapCardBrandImage[brand],style:_CardSelection2.default.cardType})),
_react2.default.createElement(_Shared.ISText,{type:'SEMIBOLD',style:_CardSelection2.default.cardName},name),
_react2.default.createElement(_Shared.ISText,{type:'SEMIBOLD',style:_CardSelection2.default.cardNumber},last_digits),
(cov_2dhks0odzo.b[2][0]++,deleteCard)&&(cov_2dhks0odzo.b[2][1]++,_react2.default.createElement(_Shared.Button,{type:'inline',text:'Delete',onPress:this.deleteCard(id),style:_CardSelection2.default.deleteButton}))));


}}]);return CardSelection;}(_react.Component);cov_2dhks0odzo.s[6]++;

CardSelection.defaultProps={
isSelected:false,
isLast:false,
cardDetails:{},
onSelect:_noop2.default};cov_2dhks0odzo.s[7]++;

CardSelection.propTypes={
deleteCard:_propTypes2.default.func,
isLast:_propTypes2.default.bool,
onSelect:_propTypes2.default.func,
cardDetails:_propTypes2.default.object,
isSelected:_propTypes2.default.bool};exports.default=

CardSelection;