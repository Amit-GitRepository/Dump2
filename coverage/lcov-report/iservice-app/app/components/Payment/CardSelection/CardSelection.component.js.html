<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Payment/CardSelection/CardSelection.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Payment/CardSelection</a> CardSelection.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/8</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">39x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">39x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">39x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* CardSelection: shows list of cards and option to pay with new card*/
&nbsp;
import jcb from '../../../assets/jcb.png';
import mastercard from '../../../assets/mastercard.png';
import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './CardSelection.style';
import visa from '../../../assets/visa.png';
import {Button, ISText, RadioButton} from '../../Shared';
import {Image, View} from 'react-native';
/* eslint-disable camelcase */
&nbsp;
const mapCardBrandImage = {
  'Visa': visa,
  'JCB': jcb,
  'Mastercard': mastercard
};
&nbsp;
class CardSelection extends Component {
&nbsp;
  deleteCard = <span class="fstat-no" title="function not covered" >(c</span>ardId) =&gt; <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    this.props.deleteCard(cardId);</span>
  }
  
  render () {
    const {isSelected, cardDetails, onSelect, isLast, deleteCard} = this.props;
    const {brand, name, last_digits, id} = cardDetails;
    return (
      &lt;View style={[styles.container, isLast &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >styles.hideBorder]</span>}&gt;
        &lt;RadioButton isSelected={isSelected} onRadioSelect={onSelect}/&gt;
        {brand &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >&lt;Image source={mapCardBrandImage[brand]} style={styles.cardType}/&gt;}</span>
        &lt;ISText type='SEMIBOLD' style={styles.cardName}&gt;{name}&lt;/ISText&gt;
        &lt;ISText type='SEMIBOLD' style={styles.cardNumber}&gt;{last_digits}&lt;/ISText&gt;
        {deleteCard &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >&lt;Button type='inline' text='Delete' onPress={this.deleteCard(id)} style={styles.deleteButton}/&gt;}</span>
      &lt;/View&gt;
    );
  }
}
CardSelection.defaultProps = {
  isSelected: false,
  isLast: false,
  cardDetails: {},
  onSelect: noop
};
CardSelection.propTypes = {
  deleteCard: PropTypes.func,
  isLast: PropTypes.bool,
  onSelect: PropTypes.func,
  cardDetails: PropTypes.object,
  isSelected: PropTypes.bool
};
export default CardSelection;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
