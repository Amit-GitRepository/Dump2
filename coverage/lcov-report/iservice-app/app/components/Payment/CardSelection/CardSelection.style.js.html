<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Payment/CardSelection/CardSelection.style.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Payment/CardSelection</a> CardSelection.style.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {colors, fonts} from '../../../themes/constants.styles';
&nbsp;
export default {
  container: {
    flexDirection: 'row',
    paddingHorizontal: 5,
    paddingVertical: 10,
    alignItems: 'center',
    borderBottomWidth: 1,
    borderColor: colors.PRIMARY_BG_SEPARATOR
  },
  cardType: {
    flex: 0,
    marginLeft: 10
  },
  cardName: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    paddingHorizontal: 5,
    marginLeft: 10
  },
  cardNumber: {
    fontSize: fonts.FONT_SIZE_NORMAL,
    color: colors.PRIMARY_ACTIONABLE
  },
  hideBorder: {
    borderBottomWidth: 0
  },
  deleteButton: {
    marginLeft: 15
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
