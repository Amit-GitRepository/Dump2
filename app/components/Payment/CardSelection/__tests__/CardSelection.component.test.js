import CardSelection from '../CardSelection.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('CardSelection component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<CardSelection onFormSubmit={jest.fn} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
