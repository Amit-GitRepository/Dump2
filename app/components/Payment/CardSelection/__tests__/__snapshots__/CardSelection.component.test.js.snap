// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSelection component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#E9E9EA",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 10,
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#F93E60",
              "borderRadius": 11,
              "borderWidth": 1,
              "height": 22,
              "justifyContent": "center",
              "width": 22,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E9E9EA",
                "borderRadius": 6,
                "height": 12,
                "width": 12,
              }
            }
          />
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#303C46",
          "fontFamily": "SukhumvitSet-SemiBold",
          "fontSize": 18,
          "marginLeft": 10,
          "maxWidth": 130,
          "paddingHorizontal": 5,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#F93E60",
          "fontFamily": "SukhumvitSet-SemiBold",
          "fontSize": 16,
        }
      }
    />
  </View>
</View>
`;
